File: Excel/427 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/427 Double Accumulative Cipher.xlsx", range = "A1:A10")
test  = read_excel("Excel/427 Double Accumulative Cipher.xlsx", range = "B1:B10")

double_accumulative_cipher = function(word) {

  result = strsplit(word, "")[[1]] %>%
    map_dbl(~match(., letters) - 1) %>%
    accumulate(~(.x + .y) %% 26) %>%
    accumulate(~(.x + .y) %% 26) %>%
    map_dbl(~. + 1) %>%
    map_chr(~letters[.]) %>%
    paste(collapse = "")

  return(result)
}

result = input %>%
  mutate(`Answer Expected` = map_chr(`Plain Text`, double_accumulative_cipher)) %>%
  select(-`Plain Text`)

identical(result, test)
#> [1] TRUE

File: Excel/400-499/427/427 Challenge.py
import pandas as pd

def double_accumulative_cipher(word):
    letters = 'abcdefghijklmnopqrstuvwxyz'
    result = ''
    accumulative_sum = 0
    for char in word:
        index = letters.index(char.lower())
        accumulative_sum = (accumulative_sum + index) % 26
        result += letters[accumulative_sum]
    return result

test = pd.read_excel("427 Double Accumulative Cipher.xlsx", usecols="B", nrows=9)

input = pd.read_excel("427 Double Accumulative Cipher.xlsx", usecols="A", nrows= 9)
input['Answer Expected'] = input['Plain Text'].apply(double_accumulative_cipher).apply(double_accumulative_cipher)
input = input.drop(columns=['Plain Text'])

print(input.equals(test))   # True